### ISSUE TITLE
implement new design for login/signup card

### ISSUE THREAD

**Issue by lmccart** [processing/p5.js-web-editor#206]
based on jerel's design:

![01 web-editor-sign-up -1024 1x](https://cloud.githubusercontent.com/assets/191056/20579092/85fb353a-b17f-11e6-857a-6abb3ef5756d.png)
![02 web-editor-log-in-1024-a 1x](https://cloud.githubusercontent.com/assets/191056/20579093/8609f110-b17f-11e6-90ba-fa31680712c8.png)
![02 web-editor-log-in-1024-b __errors 1x](https://cloud.githubusercontent.com/assets/191056/20579094/86195574-b17f-11e6-93ee-449a50069a83.png)

full set of design files here:
http://p5js.org/p5-web-editor-design.zip

**Comment by lmccart** [comment]
one more piece of documentation here:
[diagram - landing-auth-options..pdf](https://github.com/processing/p5.js-web-editor/files/610074/diagram.-.landing-auth-options.pdf)

**Comment by catarak** [comment]
I've come across an issue with this design in the process of working on #139, and I'm curious what you think:

We decided that it made sense for the site to be HTTP by default, with the option of switching to HTTPS when it makes sense (by having a button for this, see #170). However, this means that if a user is open to a sketch and not logged in, and it is HTTP, when the user clicks 'Login', in order to redirect to HTTPS the whole page will have to reload (as in, load server-side). I think this is unexpected behavior, for the whole page to reload to open a modal.

There are two ways around this, (1) load a separate page for Login/Signup/etc. and (2) to load an iframe in the modal that is HTTPS so that this design works. The second option is more time consuming to implement I think and definitely feels a bit hacky, though the first option breaks the design.

**Comment by lmccart** [comment]
Would it be crazy to change our flow to default to https, forcing redirect to https unless you are logged in? This means when the user comes (either via http or https), they are redirected (if need be) to https. If they click login, we're all set, the modals work, and they can go anywhere they want from there. If they click the switch to http button, it will alert them to login (or maybe this button wouldn't show up unless logged in).

I've just noticed a lot of cases where people don't know to use https and it stops them for a while. I can imagine fewer cases where http is required (ex: trying to load data from an http api endpoint), but these cases generally throw an error in console mentioning http vs https which would be a better clue. Compare to the case of a webcam just failing silently.

**Comment by catarak** [comment]
I'm fine with HTTPS by default. I can just keep the design as it is, and if the UX ends up being weird we can always change it. 

Happen to have any thoughts/opinions about this @shiffman?

**Comment by catarak** [comment]
Did some playing around and thinking about this:
- While it's not hard to redirect to https when not authenticated, I think that most users will interact with the site logged in, and therefore would still encounter errors when trying to https only features.
- Another option would be to redirect all users to https when they first enter the site, and then have a UI button to allow them to switch to http. Problem is, I'm not sure how to distinguish users first entering the site vs purposefully loading the whole site again in http. There's maybe some way with cookies or query string parameters but this seems bug prone. 
- I tried locally redirecting from http to https when navigating to `/login` or `/signup` and to me it feels like the site is broken. I think it's especially important for the parts of the site where users are entering passwords and stuff to feel secure. 

IN CONCLUSION, I think the best way to solve this is to render the authentication flow views on separate pages, as it was when I first implemented this stuff but with the new design. But obviously I'm open to other solutions if we think of them.

Who knew this would be so hard!

**Comment by shiffman** [comment]
@catarak all sounds great to me!  Definitely ok with defaulting more towards https if that helps smooth out kinks.

**Comment by lmccart** [comment]
Hm I see the issues, thanks for breaking it down. Let's go with your conclusion.

**Comment by lmccart** [comment]
the design has been adapted to work with the HTTPS plan outlined above, it has been implemented so I think we're all set here!
